protocol/messenger_handler.go

```diff
@@ -1442,7 +1442,41 @@ func (m *Messenger) HandleCommunityCancelRequestToJoin(state *ReceivedMessageSta
 		return ErrInvalidCommunityID
 	}
 
-	requestToJoin, err := m.communitiesManager.HandleCommunityCancelRequestToJoin(signer, cancelRequestToJoinProto)
+	community, err := m.communitiesManager.GetByID(cancelRequestToJoinProto.CommunityId)
+
+	if err != nil {
+		return err
+	}
+
+	senderID := state.CurrentMessageState.Contact.ID
+	senderPubKey, err := common.HexToPubkey(senderID)
+
+	if err != nil {
+		return err
+	}
+
+	previousRequestToJoin, err := m.communitiesManager.GetRequestToJoinByPkAndCommunityID(senderPubKey, community.ID())
+
+	if err != nil {
+		return err
+	}
+
+	if community.HasMember(senderPubKey) && cancelRequestToJoinProto.Clock > previousRequestToJoin.Clock {
+		response, err := m.RemoveUserFromCommunity(
+			community.ID(),
+			senderID,
+		)
+		if err != nil {
+			return err
+		}
+
+		if len(response.Communities()) > 0 {
+			state.Response.AddCommunity(response.Communities()[0])
```

Comment:
Wouldn't we want the indicator that a member has been removed in the response?

Reply:
Oh I see. So you are adding the community to mean "member removed"? If that's the case then I think it's really hard to grasp this because it's not obvious why having a community in the response is related.

But do we really need an indicator that the member has been removed? Do you mean the clients will use this information? Because if clients are not using this, than no code is better than any code we would come up with.

Reply:
I think this response returns to the admin/privliged user of the community, And I think it might be necessary to let the admin client know someone is no longer a member, But it's just an assumption.
WDYT Icaro?

Reply:
Got it, Will do.
Thanks Icaro!